riscv/processor.cc

```diff
@@ -397,6 +397,33 @@ void state_t::reset(processor_t* const proc, reg_t max_isa)
   const reg_t henvcfg_init = (proc->extension_enabled(EXT_SVPBMT) ? HENVCFG_PBMTE : 0);
   csrmap[CSR_HENVCFG] = henvcfg = std::make_shared<henvcfg_csr_t>(proc, CSR_HENVCFG, henvcfg_mask, henvcfg_init, menvcfg);
 
+  const reg_t mstateen0_mask = (proc->extension_enabled(EXT_ZFINX) ? MSTATEEN0_FCSR : 0) |
+                               MSTATEEN0_CS | MSTATEEN0_HENVCFG | MSTATEEN_HSTATEEN;
+  const reg_t mstateen0h_mask = MSTATEEN0H_HENVCFG | MSTATEENH_HSTATEEN;
```

Comment:
Since the high32 CSRs are reading & writing through the full 64-bit CSR, do they really need their own mask? Can you make the mask -1 for these? (`mstatush` might be a special case because the fields are slightly different than RV64.)

Reply:
Yeah. It's OK currently.

Reply:
My concern is that the `mstaten0_mask` and `mstateen0h_mask` could easily get out of sync as new bits are added to the spec. As I understand it, RV32's `mstateen0h` will always be identical to the upper 32 bits of RV64's `mstateen`, so by using -1 as the mask, it will avoid any possibility of the two masks getting unsynchronized in the future.